### DISCUSSION TITLE
Cant increase socket timeout

### DISCUSSION THREAD

**Discussion by ala85** [danny-avila/LibreChat#4586] [Troubleshooting]
### What happened?

When using Bedrock via AWS credentials behind proxy, I need to extend socket time out as getting

TimeoutError: Socket timed out without establishing a connection within 3100 ms
    at Timeout._onTimeout (app\node_modules\@smithy\node-http-handler\dist-cjs\index.js:67:21)
    at listOnTimeout (node:internal/timers:581:17)
    at process.processTimers (node:internal/timers:519:7)

Is there a means to increase the value?

### Steps to Reproduce

1. Use Bedrock
2. Set AWS Cderentials in env file
3. Need behind a proxy

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
TimeoutError: Socket timed out without establishing a connection within 3100 ms
    at Timeout._onTimeout (app\node_modules\@smithy\node-http-handler\dist-cjs\index.js:67:21)
    at listOnTimeout (node:internal/timers:581:17)
    at process.processTimers (node:internal/timers:519:7)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
This is a niche use case, try seeing if you can pass a BedrockRuntimeClient to the chat model class from `@langchain/aws`, which LibreChat uses for Bedrock support, in a one-off script to see if it's even possible: https://js.langchain.com/docs/integrations/chat/bedrock_converse

**Reply by danny-avila** [reply]
I believe you can pass `client` as a field to the constructor, like the `model` field, to achieve something like this. Take a look at what Claude recommends: https://demo.librechat.ai/share/HCHLd3KsCe8H12RNO47Up

**Reply by ala85** [reply]
Could you elaborate where the client should be passed? To which class constructor?

**Reply by ala85** [reply]
Ok, found out. I ll open PR to @langchain/aws. Once released there can we update its version in @librechat/agents and/or this repository? @danny-avila

**Reply by danny-avila** [reply]
yes thank you

**Reply by ala85** [reply]
Hi @danny-avila @langchain/aws@0.1.2 now contains the change. Where can this be applied in the librechat/agents repo?

**Reply by ala85** [reply]
@danny-avila any hint on updating langchain/aws to the librechat/agent library?

**Reply by ala85** [reply]
@berry-13 would you know how to increase above dependency?

**Reply by berry-13** [reply]
> @berry-13 would you know how to increase above dependency?

unfortunately, I can't modify the LibreChat's Agents package

**Reply by ala85** [reply]
@danny-avila would be possible for you?

**Reply by danny-avila** [reply]
I will bump the package soon, thanks

**Reply by ala85** [reply]
@danny-avila was this bumped by any chance? Thanks

**Reply by danny-avila** [reply]
yes

**Reply by ala85** [reply]
Thanks @danny-avila  out of curiosity, could you share PR or where this change was done? Is it an internal package?

**Reply by danny-avila** [reply]
It's an internal package. Latest version is 0.1.2 according to https://www.npmjs.com/package/@langchain/aws

https://github.com/danny-avila/LibreChat/blob/51e016ef2c2de7303093f616822dcfff165cea7d/package-lock.json#L690
